webui-src/app/config/config_mail.js

```diff
@@ -0,0 +1,188 @@
+const m = require('mithril');
+const rs = require('rswebui');
+const widget = require('widgets');
+const util = require('config/config_util');
+
+const msgTagObj = {
+  tagId: 100,
+  tagName: '',
+  tagColor: '',
+};
+let tagArr = [];
+
+async function handleSubmit(tagId) {
+  const modalContainer = document.getElementById('modal-container');
+  msgTagObj.tagId = typeof tagId === 'number' ? tagId : util.getRandomId(tagArr);
+  let tagNameAlreadyExists = false;
+  tagArr.forEach((item) => {
+    if (item.value.first === msgTagObj.tagName) tagNameAlreadyExists = true;
+  });
+  if (tagNameAlreadyExists) {
+    alert('Tag Name Already Exists');
+  } else {
+    rs.rsJsonApiRequest('/rsMsgs/setMessageTagType', {
+      tagId: msgTagObj.tagId,
+      text: msgTagObj.tagName,
+      rgb_color: parseInt(msgTagObj.tagColor.substring(1), 16),
+    });
+    modalContainer.style.display = 'none';
+    rs.rsJsonApiRequest('/rsMsgs/getMessageTagTypes').then((res) => (tagArr = res.body.tags.types));
+  }
+}
+
+const MessageTagForm = () => {
+  return {
+    view: (v) => {
+      const isCreateForm = v.attrs.tagItem === undefined;
+      return m(
+        'form.mail-tags-form',
+        {
+          onsubmit: isCreateForm ? handleSubmit : () => handleSubmit(v.attrs.tagItem.key),
+        },
+        [
+          m('h3', isCreateForm ? 'Create New Tag Type' : 'Edit Tag Type'),
+          m('hr'),
+          m('.input-field', [
+            m('label[for=tagName]', 'Enter Tag Name'),
+            m('input[type=text][id=tagName][placeholder="enter tag name"]', {
+              value: msgTagObj.tagName,
+              oninput: (e) => (msgTagObj.tagName = e.target.value),
+            }),
+          ]),
+          m('.input-field', [
+            m('label[for=tagColor]', 'Choose Tag Color'),
+            m('input[type=color][id=tagColor]', {
+              value: msgTagObj.tagColor,
+              oninput: (e) => (msgTagObj.tagColor = e.target.value),
+            }),
+          ]),
+          // v.attrs.tagItem !== undefined && m('p', v.attrs.tagItem.value.first),
+          m('button[type=submit]', 'Submit'),
+        ]
+      );
+    },
+  };
+};
+
+const Mail = () => {
+  let distantMessagingPermissionFlag = 0;
+  return {
+    oninit: () => {
+      rs.rsJsonApiRequest('/rsMsgs/getMessageTagTypes').then(
+        (res) => (tagArr = res.body.tags.types)
+      );
+      rs.rsJsonApiRequest('/rsMsgs/getDistantMessagingPermissionFlags').then(
+        (res) => (distantMessagingPermissionFlag = res.body.retval)
+      );
+    },
+    view: () =>
+      m('.widget.widget-half.mail', [
+        m('.mail-tags__heading', m('h3', 'Distant Messages')),
+        m('hr'),
+        m('.permission-flag', [
+          m('p', 'Accept encrypted distant messages from: '),
+          m(
+            'select[id=setDistantMessagingPermission]',
```

Comment:
Do we need id here?

Reply:
I thought that I would do it to add some styles, but I didn't need to do anything and forgot to remove it.
Will do it now. 👍🏼